src/native/external/zlib-ng/PORTING.md

```diff
@@ -4,6 +4,38 @@ Porting applications to use zlib-ng
 Zlib-ng can be used/compiled in two different modes, that require some
 consideration by the application developer.
 
+Changes from zlib affecting native and compat modes
+---------------------------------------------------
+Zlib-ng is not as conservative with memory allocation as Zlib is.
+
+Where Zlib's inflate will allocate a lower amount of memory depending on
+compression level and window size, zlib-ng will always allocate the maximum
+amount of memory and possibly leave parts of it unused.
+Zlib-ng's deflate will however allocate a lower amount of memory depending
+on compression level and window size.
+
+Zlib-ng also allocates one "big" buffer instead of doing multiple smaller
```

Comment:
Is the custom wrapper of the memory allocator that inserts additional padding still relevant given this change?

If we keep the custom wrapper of the memory allocator, we need to check the performance impact of the buffer zero initialization that it does.

Reply:
I'll collect the perf comparisons.